datasets/swda/swda.py

```diff
@@ -169,35 +423,356 @@ def _split_generators(self, dl_manager):
         This method is tasked with downloading/extracting the data and defining the splits.
         """
 
+        dl_dir = dl_manager.download_and_extract(_URL)
+        data_dir = os.path.join(dl_dir, "swda")
+
         urls_to_download = self._URLS
         downloaded_files = dl_manager.download_and_extract(urls_to_download)
 
         return [
             datasets.SplitGenerator(
-                name=datasets.Split.TRAIN, gen_kwargs={"filepath": downloaded_files["train"], "split": "train"}
+                name=datasets.Split.TRAIN, gen_kwargs={"data_dir": data_dir, "split_file": downloaded_files["train"]}
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.VALIDATION,
+                gen_kwargs={"data_dir": data_dir, "split_file": downloaded_files["dev"]},
             ),
             datasets.SplitGenerator(
-                name=datasets.Split.VALIDATION, gen_kwargs={"filepath": downloaded_files["dev"], "split": "validation"}
+                name=datasets.Split.TEST, gen_kwargs={"data_dir": data_dir, "split_file": downloaded_files["test"]}
             ),
         ]
 
-    def _generate_examples(self, filepath, split):
+    def _generate_examples(self, data_dir, split_file):
         """
         Yields examples.
         This method will receive as arguments the `gen_kwargs` defined in the previous `_split_generators` method.
         It is in charge of opening the given file and yielding (key, example) tuples from the dataset
         The key is not important, it's more here for legacy reason (legacy from tfds).
         """
 
-        with open(filepath, encoding="utf-8") as f:
-            for id_, row in enumerate(f):
+        # Read in the split file.
+        split_file = io.open(file=split_file, mode="r", encoding="utf-8").read().splitlines()
+        # Read in corpus data using split files.
+        corpus = CorpusReader(src_dirname=data_dir, split_file=split_file)
+        # Generate examples.
+        for i_trans, trans in enumerate(corpus.iter_transcripts()):
+            for i_utt, utt in enumerate(trans.utterances):
+                id_ = i_trans + i_utt
+                yield id_, {
+                    "swda_filename": utt.swda_filename,
+                    "ptb_basename": utt.ptb_basename,
+                    "conversation_no": utt.conversation_no,
+                    "transcript_index": utt.transcript_index,
+                    "act_tag": utt.act_tag,  # Need change to labels
```

Comment:
Why `Need change to labels` ?

Reply:
wops, that was just an old comment, sorry 😄